pubsub/src/main/scala/org/byrde/pubsub/AdminClientTrait.scala

```diff
@@ -0,0 +1,54 @@
+package org.byrde.pubsub
+
+import com.google.api.gax.core.{CredentialsProvider, NoCredentialsProvider}
+import com.google.api.gax.grpc.GrpcTransportChannel
+import com.google.api.gax.rpc.{FixedTransportChannelProvider, TransportChannelProvider}
+import com.google.cloud.pubsub.v1.{SubscriptionAdminClient, SubscriptionAdminSettings, TopicAdminClient, TopicAdminSettings}
+import io.grpc.{ManagedChannel, ManagedChannelBuilder}
+import org.byrde.logging.Logger
+
+trait AdminClientTrait {
+  private var channel: ManagedChannel = _
+  private def transportChannel(maybeHost: Option[String])(implicit logger: Logger): Option[TransportChannelProvider] =
```

Comment:
we need to create this provider only once and share it

Reply:
maybe put it as a lazy val then?